compiler-plugin/src/main/kotlin/arrow/meta/quotes/ClassOrObject.kt

```diff
@@ -13,12 +13,112 @@ import org.jetbrains.kotlin.psi.psiUtil.getValueParameters
 import org.jetbrains.kotlin.psi.psiUtil.modalityModifierType
 import org.jetbrains.kotlin.psi.psiUtil.visibilityModifierType
 
+/**
+ * [classOrObject] is a function that intercepts all [KtClass] elements that [match]
+ * allowing a [Transform] to change the intercepted AST tree before compilation.
+ *
+ * An extension function of [Meta] and inheriting from [ExtensionPhase], [classOrObject] was designed to to feed in
+ * virtually any kind of [KtClass] predicate, followed by mapping a function that takes the desired [Scope] of our
+ * [KtClass] to change whatever PSI elements desired.
+ *
+ * For example, the [LensPlugin] and the [HigherKindPlugin] favor easy update of immutable data structures and ad-hoc polymorphism that does not require inheritance.
+ *
+ * In the [LensPlugin], we check to see if a type is a product type.  To add context, the cardinality of a product
+ * type is the product of its contained properties:
```

Comment:
nit: Is this newline intended? to be honest I get a bit lost on where the newlines within paragraphs should be added or whether they shouldn't at all, given they'll break the text in random places depending on how it's rendered on the site. Do we have an strategy for that? Should we go with a one liner for complete paragraphs so the site breaks it at will when it reaches the end? cc @raulraja @pakoito

Reply:
I guess just making it look good when running the site.

Reply:
I'll be sure to check locally for the next push!

Reply:
I have them! Thanks!